app/models/pubmed_source_record.rb

```diff
@@ -68,6 +68,19 @@ def self.get_and_store_records_from_pubmed(pmids)
     PubmedSourceRecord.import source_records
   end
 
+  # Retrieve this pubmed record from PubMed and update
+  # is_active, source_data and the source_fingerprint fields.
+  # Used to update the pubmed source record on our end (similar to .sciencewire_update)
+  # @return [Boolean] the return value from update_attributes!
+  def pubmed_update
+    pubmed_source_xml = PubmedClient.new.fetch_records_for_pmid_list pmid
+    pub_doc = Nokogiri::XML(pubmed_source_xml).xpath('//PubmedArticle')[0]
+    attrs = {}
+    attrs[:source_data] = pub_doc.to_xml
+    attrs[:source_fingerprint] = Digest::SHA2.hexdigest(pub_doc)
+    update_attributes! attrs
+  end
```

Comment:
This looks OK.  Just a few thoughts or questions arise from this:
- this seems to use a high-level command that accepts a list of PMID values, or a single value is OK too?
- is there any Pubmed HEAD request that contains cache invalidation data that we can track to detect updates?  If there is, do we already have the data in our PubmedSourceRecord data (if not, then it's not going to work, unless we start to add it).  If there is, we could run a rake task over all the Pubmed data, say monthly or something (no less than weekly because their update schedule is weekly anyway).